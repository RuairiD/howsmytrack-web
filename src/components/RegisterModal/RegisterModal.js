import React from 'react';

import { Modal } from 'antd';
import RegisterForm from '../RegisterForm/RegisterForm';

type Props = {
    onCancel: () => void,
    isVisible: boolean,
};

class RegisterModal extends React.Component<Props> {
    /*
     * Component without docs
     */
    render() {
        return (
            <Modal
                title="Register"
                onCancel={this.props.onCancel}
                visible={this.props.isVisible}
                footer={null}
                destroyOnClose
            >
                <RegisterForm />
            </Modal>
        );
    }
}

export default RegisterModal;
