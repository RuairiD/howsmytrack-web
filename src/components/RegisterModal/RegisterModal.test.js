import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import RegisterModal from "./RegisterModal";

describe("RegisterModal", () => {
    it("renders a Modal containing a RegisterForm", () => {
        const onCancel = () => {};
        const wrapper = shallow(
            <RegisterModal
                onCancel={onCancel}
                isVisible
            />,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
