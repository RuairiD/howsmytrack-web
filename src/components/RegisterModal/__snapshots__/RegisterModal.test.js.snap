// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterModal renders a Modal containing a RegisterForm 1`] = `
<Modal
  className="responsive-modal"
  confirmLoading={false}
  destroyOnClose={true}
  footer={null}
  maskTransitionName="fade"
  okType="primary"
  onCancel={[Function]}
  title="Register"
  transitionName="zoom"
  visible={true}
  width={520}
>
  <Form(UnwrappedRegisterFormController) />
</Modal>
`;
